flow/beta/cli/config/json/account.go

```diff
@@ -0,0 +1,200 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package json
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/onflow/flow-cli/flow/beta/cli/config"
+	"github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go-sdk/crypto"
+)
+
+type jsonAccounts map[string]jsonAccount
+
+// transformChainID return chain id based on address and chain id
+func transformChainID(rawChainID string, rawAddress string) flow.ChainID {
+	if rawAddress == "service" && rawChainID == "" {
+		return flow.Emulator
+	}
+	return flow.ChainID(rawChainID)
+}
+
+// transformAddress returns address based on address and chain id
+func transformAddress(rawAddress string, rawChainID string) flow.Address {
+	var address flow.Address
+	chainID := transformChainID(rawChainID, rawAddress)
+
+	if rawAddress == "service" {
+		address = flow.ServiceAddress(chainID)
+	} else {
+		rawAddress = strings.ReplaceAll(rawAddress, "0x", "") // remove 0x if present
+		address = flow.HexToAddress(rawAddress)
+	}
+
+	return address
+}
+
+// transformToConfig transforms json structures to config structure
+func (j jsonAccounts) transformToConfig() config.Accounts {
+	accounts := make(config.Accounts, 0)
+
+	for accountName, a := range j {
+		var account config.Account
+		// simple format
+		if a.Simple.Address != "" {
+			account = config.Account{
+				Name:    accountName,
+				ChainID: transformChainID(a.Simple.Chain, a.Simple.Address),
+				Address: transformAddress(a.Simple.Address, a.Simple.Chain),
+				Keys: []config.AccountKey{{
+					Type:     config.KeyTypeHex,
+					Index:    0,
+					SigAlgo:  crypto.ECDSA_P256,
+					HashAlgo: crypto.SHA3_256,
+					Context: map[string]string{
+						"privateKey": a.Simple.Keys,
+					},
+				}},
+			}
+		} else { // advanced format
+			keys := make([]config.AccountKey, 0)
+			for _, key := range a.Advanced.Keys {
+				key := config.AccountKey{
+					Type:     key.Type,
+					Index:    key.Index,
+					SigAlgo:  crypto.StringToSignatureAlgorithm(key.SigAlgo),
+					HashAlgo: crypto.StringToHashAlgorithm(key.HashAlgo),
+					Context:  key.Context,
+				}
+				keys = append(keys, key)
+			}
+
+			account = config.Account{
+				Name:    accountName,
+				ChainID: transformChainID(a.Advanced.Chain, a.Advanced.Address),
+				Address: transformAddress(a.Advanced.Address, a.Advanced.Chain),
+				Keys:    keys,
+			}
+		}
+
+		accounts = append(accounts, account)
+	}
+
+	return accounts
+}
+
+// transformToJSON transforms config structure to json structures for saving
+func (j jsonAccounts) transformToJSON(accounts config.Accounts) jsonAccounts {
+	jsonAccounts := jsonAccounts{}
+
+	for _, a := range accounts {
+		// if simple
+		//REF: horible implementation improvemenet needed
+		if len(a.Keys) == 1 && a.Keys[0].Index == 0 &&
+			a.Keys[0].Type == config.KeyTypeHex &&
+			a.Keys[0].SigAlgo == crypto.ECDSA_P256 &&
+			a.Keys[0].HashAlgo == crypto.SHA3_256 {
```

Comment:
Yeah, I was thinking about how we'd reverse this case. Given just a key struct, there's technically no way of knowing if it should be reduced to the simple case.

I think this is the right approach, though. Maybe just move this if clause to a separate helper function? e.g. `isDefaultKeyFormat(a.Keys)`